packages/react/src/components/header/components/headerNavigationMenu/HeaderNavigationMenu.tsx

```diff
@@ -38,23 +38,20 @@ export const HeaderNavigationMenu = ({ ariaLabel, children, id }: HeaderNavigati
       <ul className={styles.headerNavigationMenuList}>
         <HeaderNavigationMenuContext.Provider value={context}>
           {Children.map(childElements, (child, index) => {
-            if (React.isValidElement(child)) {
-              const linkContainerClasses = child.props.active
-                ? classNames(styles.headerNavigationMenuLinkContent, styles.headerNavigationMenuLinkContentActive)
-                : styles.headerNavigationMenuLinkContent;
-              return (
-                // eslint-disable-next-line react/no-array-index-key
-                <li key={index} className={styles.headerNavigationMenuLinkContainer}>
-                  <span className={linkContainerClasses}>
-                    {cloneElement(child as React.ReactElement, {
-                      className: classNames(child.props.className, styles.headerNavigationMenuLink),
-                      index,
-                    })}
-                  </span>
-                </li>
-              );
-            }
-            return null;
+            const linkContainerClasses = child.props.active
+              ? classNames(styles.headerNavigationMenuLinkContent, styles.headerNavigationMenuLinkContentActive)
+              : styles.headerNavigationMenuLinkContent;
+            return (
+              // eslint-disable-next-line react/no-array-index-key
+              <li key={index} className={styles.headerNavigationMenuLinkContainer}>
+                <span className={linkContainerClasses}>
+                  {cloneElement(child, {
```

Comment:
Should we have this element cloning logic done when the children are saved to the context to prevent clones being made multiple times? For example inside the getChildElementsEvenIfContainerInbetween function.

Reply:
Sounds like a good idea to prevent multiple cloning. But I think that the whole `{Children.map(...` usage should be removed because of the legacy thing. Maybe Header.NavigationLink will have `variant` prop as well? I mentioned Children API refactoring in [this here ticket](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1578).